legend-engine-authentication-pure/src/main/resources/core_authentication/metamodel/metamodel.pure

```diff
@@ -0,0 +1,57 @@
+// Copyright 2021 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import meta::pure::authentication::*;
+
+Class <<typemodifiers.abstract>> meta::pure::authentication::AuthenticationSpec
+{
+
+}
+
+Class meta::pure::authentication::OauthAuthenticationSpec extends meta::pure::authentication::AuthenticationSpec
+{
+    grantType: String[1];
+    clientId: String[1];
+    clientSecretVaultReference: String[0..1];
+    authServerUrl: String[1];
+}
+
+Class meta::pure::authentication::UsernamePasswordAuthenticationSpec extends meta::pure::authentication::AuthenticationSpec
+{
+   username: String[1];
+   password: VaultCredential[1];
+}
+
+Enum meta::pure::authentication::Location
+{
+  header,
+  cookie
+}
+
+Class meta::pure::authentication::ApiKeyAuthenticationSpec extends meta::pure::authentication::AuthenticationSpec
```

Comment:
Is ApiKeyAuthenticationSpec specific to ServiceStore? Looking location Enum it seems to be the case

Reply:
> Is ApiKeyAuthenticationSpec specific to ServiceStore? Looking location Enum it seems to be the case

No it is not. For e.g Databricks connector uses Api key authentication